examples/webgl_buffergeometry_instancing_lambert.html

```diff
@@ -297,71 +220,20 @@
 			geometry.addAttribute( 'instanceColor', new THREE.InstancedBufferAttribute( colors, 3 ) );
 			geometry.addAttribute( 'instanceScale', new THREE.InstancedBufferAttribute( scales, 1 ) );
 
-
 			// material
 
 			var envMap = new THREE.TextureLoader().load( `textures/metal.jpg`, function ( texture ) {
 
 				texture.mapping = THREE.SphericalReflectionMapping;
-
+				material.needsUpdate = true
 			} );
 
-			var shader = THREE.CustomShaderLib[ 'lambert' ];
-
-			var uniforms = THREE.UniformsUtils.clone( shader.uniforms );
-
-			uniforms[ "diffuse" ].value.set( 0xffb54a );
-			uniforms[ "envMap" ].value = envMap;
-			uniforms[ "reflectivity" ].value = 1;
+			material.envMap = envMap
+			material.needsUpdate = true
 
 			// defines - Since we are reusing the ShaderChunks, we must specify the required defines.
 			// The renderer does not set these defines for ShaderMaterial
 
-			var defines = {
```

Comment:
`MeshLambertMaterial` may be obscuring the defines as a very low level concept (it's not documented and user is not aware of it). This is essentially an extension of `MeshLambertMaterial` however it requires the low-level defines to be manually provided. One has to be familiar with all of these names and what they do. 

Only one particular flag here is actually relevant for the example of instancing. The rest are relevant to the rewiring/mimicking with the `THREE.ShaderMaterial`